source/characterProcessing.py

```diff
@@ -402,82 +405,56 @@ def _saveSymbol(self, symbol):
 		return "\t".join(fields)
 
 
-_noSymbolLocalesCache = set()
-_noCLDRLocalesCache = set()
-
-
-def _getSpeechSymbolsForLocale(locale: str) -> Tuple[SpeechSymbols, SpeechSymbols]:
-	if locale in _noSymbolLocalesCache and (
-		locale in _noCLDRLocalesCache or not config.conf["speech"]["includeCLDR"]
-	):
-		raise LookupError
-	builtinDataImported = False
-	builtin = SpeechSymbols()
-	if config.conf["speech"]["includeCLDR"]:
-		# Try to load CLDR data when processing is on.
-		# Load the data before loading other symbols,
-		# in order to allow translators to override them.
+def _getSpeechSymbolsForLocale(locale: str) -> list[SpeechSymbols]:
+	symbols: list[SpeechSymbols] = []
+	for definition in getAvailableSymbolDictionaryDefinitions():
+		if not definition.enabled:
+			continue
 		try:
-			builtin.load(
-				os.path.join(globalVars.appDir, "locale", locale, "cldr.dic"),
-				allowComplexSymbols=False,
+			symbols.append(definition.getSymbols(locale))
+		except (LookupError, FileNotFoundError):
+			log.debugWarning(
+				f"Error loading {definition.name!r} symbols for locale {locale!r}",
+				exc_info=True,
 			)
-			builtinDataImported = True
-		except IOError:
-			_noCLDRLocalesCache.add(locale)
-			log.debugWarning("No CLDR data for locale %s" % locale)
-	try:
-		builtin.load(os.path.join(globalVars.appDir, "locale", locale, "symbols.dic"))
-		builtinDataImported = True
-	except IOError:
-		_noSymbolLocalesCache.add(locale)
-		log.debugWarning("No symbol data for locale %s" % locale)
-	if not builtinDataImported:
-		raise LookupError("No symbol information for locale %s" % locale)
-	user = SpeechSymbols()
-	pathToSymbolsDic = WritePaths.getSymbolsConfigFile(locale)
-	try:
-		# Don't allow users to specify complex symbols
-		# because an error will cause the whole processor to fail.
-		user.load(pathToSymbolsDic, allowComplexSymbols=False)
-	except IOError:
-		# An empty user SpeechSymbols is okay.
-		pass
-	return builtin, user
+			return symbols
```

Comment:
Good catch. This might explain why @paulGeoghegan was missing symbols here and there